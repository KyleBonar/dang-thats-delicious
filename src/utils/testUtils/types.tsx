export interface MockLink {
  text: string;
  to: string;
  target: "_blank" | "_self";
}
