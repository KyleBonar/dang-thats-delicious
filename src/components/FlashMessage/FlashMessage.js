import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { flashClose } from "../../actions/flash";

import Checker from "../../helper/Checker/Checker.js";

const FlashMessage = ({ flashMessage, flashClose }) => {
  function iterateObject() {
    // the keys of this.state.text may change later depending on
    //the DB and this way of looping through the object will us to grab the messages
    //reguardless of what the key is
    // .reverse() to get error message in correct order of form layout
    const errorKeys = Object.keys(flashMessage.text).reverse();
    const errorMessages = errorKeys.map(key => {
      return flashMessage.text[key].message;
    });
    return errorMessages.map((errorMessage, index) => {
      return (
        <p key={index} className="item">
          {errorMessage}
        </p>
      );
    });
  }

  function iterateArray() {
    return flashMessage.text.map((txt, index) => {
      return (
        <p key={index} className="item">
          {txt}
        </p>
      );
    });
  }

  const slugLink = flashMessage.slug ? (
    <Link to={`/store/${flashMessage.slug}`}>Rate it!</Link>
  ) : (
    ""
  );

  const { text, type } = flashMessage;

  return (
    <div className={`flash ${type}`}>
      {/*if object, iterate over object*/}
      {Checker.isObject(text) && (
        <div className="error-list">{this.iterateObject()}</div>
      )}

      {/*if array, iterate over array*/}
      {Checker.isArray(text) && (
        <div className="error-list">{this.iterateArray()}</div>
      )}

      {/*if string, output string*/}
      {Checker.isString(text) && (
        <div className="success-list">
          <p className="item">
            {text} {slugLink}
          </p>
        </div>
      )}
      <button className="close-button" onClick={flashClose}>
        X
      </button>
    </div>
  );
};

FlashMessage.propTypes = {
  flashMessage: PropTypes.shape({
    isVisible: PropTypes.bool.isRequired,
    type: PropTypes.string.isRequired,
    text: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
      PropTypes.object
    ])
  }).isRequired,
  flashClose: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    flashMessage: state.flashMessage
  };
}

export default connect(mapStateToProps, { flashClose })(FlashMessage);
