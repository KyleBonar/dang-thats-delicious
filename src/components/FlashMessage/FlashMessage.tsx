import * as React from "react";
import styled from "styled-components";

const StyledContainer = styled.div`
  display: flex;
  flex-direction: row;

  width: 100%;
  margin-bottom: 15px;
`;

export interface FlashMessageProps {
  className?: string;
  type?: "success" | "warning" | "alert";
  isVisible: boolean;
  slug?: string;
  text?: string;
  children?: string;
}

const FlashMessage: React.SFC<FlashMessageProps> = props => {
  return (
    <StyledContainer className={props.className}>
      {props.isVisible && <div>{props.children || props.text}</div>}
    </StyledContainer>
  );
};

FlashMessage.defaultProps = {
  isVisible: true
};

const StyledFlashMessage = styled(FlashMessage)`
  div {
    padding: 10px 15px;
    width: 100%;
    box-sizing: border-box;
    background-color: ${props =>
      props.type === "success"
        ? "#dff0d8"
        : props.type === "warning"
        ? "#fcf8e3"
        : "#f2dede"};
    border-left: 7px solid
      ${props =>
        props.type === "success"
          ? "#d0e9c6"
          : props.type === "warning"
          ? "#faf2cc"
          : "#ebcccc"};
    color: ${props =>
      props.type === "success"
        ? "#3c763d"
        : props.type === "warning"
        ? "#8a6d3b"
        : "#a94442"};
  }
`;

export { StyledFlashMessage as FlashMessage };
