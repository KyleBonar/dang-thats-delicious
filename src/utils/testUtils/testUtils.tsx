import * as React from "react";
import * as enzyme from "enzyme";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";
import { IErrReturn } from "../Err/Err";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.random_element([undefined, casual.string]),
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.random_element([undefined, casual.array_of_words()]),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const generateFakeSauces = (): ISauce[] =>
  new Array(casual.integer(1, 15)).fill(null).map(fakeSauce);

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = generateFakeSauces();
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  const featured: undefined | string[] = casual.random_element([
    undefined,
    allSlugs.length > 0
      ? allSlugs.filter(slug => {
          if (casual.random < 0.4) return slug;
          return;
        })
      : undefined
  ]);

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const middlewares = [thunk];
  const storeConfig = configureStore(middlewares);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(1, MIN_PASSWORD_LENGTH - 1);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

// Function designed to resemble how a network error would look like.
// This will help the catch(err) only have to handle a single format
const generateErr = (): IErrReturn => {
  return {
    response: {
      data: {
        status: 400,
        msg: casual.string,
        isGood: false
      }
    }
  };
};

// lets us mount up a simple custom hook
const mountReactHook = hook => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.shallow(
      <Component>
        {hookValues => {
          Object.assign(componentHook, hookValues);
          return null;
        }}
      </Component>
    );
  });
  return { componentMount, componentHook };
};

export interface ImountReactHookWithReduxStore {
  componentMount: any;
  componentHook: any;
}
const mountReactHookWithReduxStore = (
  hook,
  store
): ImountReactHookWithReduxStore => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.mount(
      <Provider store={store}>
        <Component>
          {hookValues => {
            Object.assign(componentHook, hookValues);
            return null;
          }}
        </Component>
      </Provider>
    );
  });
  return { componentMount, componentHook };
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword,
  generateInValidPassword,
  generateErr,
  mountReactHook,
  mountReactHookWithReduxStore
};
