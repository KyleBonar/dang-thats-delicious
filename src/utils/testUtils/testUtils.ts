import * as React from "react";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";

import { MockCard } from "./types";
import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import { FlashMessageProps } from "../../components/FlashMessage/FlashMessage";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.string,
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.array_of_words(),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = new Array(casual.integer(1, 15))
    .fill(null)
    .map(fakeSauce);
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured: undefined,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const storeConfig = configureStore([]);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const fakeCard = (): MockCard => ({
  title: casual.string,
  description: casual.string,
  to: casual.url,
  showLink: casual.boolean,
  imageLink: casual.url,
  className: casual.string,
  anchorText: casual.string
});

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(0, MIN_PASSWORD_LENGTH);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword
};
