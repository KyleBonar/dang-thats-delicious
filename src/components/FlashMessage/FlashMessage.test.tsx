import * as React from "react";
import * as enzyme from "enzyme";

import { casual } from "../../utils/testUtils/testUtils";
import { FlashMessage, FlashMessageProps } from "./FlashMessage";

const mockFlashMessage = (): FlashMessageProps => ({
  className: casual.random_element([undefined, casual.string]),
  type: casual.random_element([undefined, "success", "warning", "alert"]),
  isVisible: casual.boolean,
  slug: casual.random_element([undefined, casual.url]),
  slugText: casual.random_element([undefined, casual.name]),
  text: casual.random_element([undefined, casual.string]),
  children: casual.random_element([undefined, casual.string])
});

describe("<FlashMessage />", () => {
  it("renders", () => {
    const wrapper = enzyme.shallow(<FlashMessage {...mockFlashMessage()} />);

    expect(wrapper).toBeTruthy();
  });
});
