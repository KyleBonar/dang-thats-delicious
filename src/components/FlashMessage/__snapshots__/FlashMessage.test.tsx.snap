// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlashMessage /> matches snapshot 1`] = `""`;

exports[`<FlashMessage /> matches snapshot 2`] = `
<styled.div>
  <styled.div>
    <styled.div>
      animi vel provident dicta perferendis dolorum cupiditate
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 3`] = `
<styled.div
  className="saepe ut assumenda sint quia sed exercitationem"
>
  <styled.div>
    <styled.div>
      debitis rem similique temporibus aliquam sed sapiente
       
      <Link
        href="http://Klocko.info/"
      >
        Mrs. Keanu Weissnat
      </Link>
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 4`] = `""`;

exports[`<FlashMessage /> matches snapshot 5`] = `""`;

exports[`<FlashMessage /> matches snapshot 6`] = `
<styled.div
  className="eveniet assumenda quis quae iusto sapiente nostrum"
>
  <styled.div>
    <styled.div>
      et maxime quia dolores quisquam aut consequatur
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 7`] = `""`;

exports[`<FlashMessage /> matches snapshot 8`] = `
<styled.div>
  <styled.div>
    <styled.div>
      autem ut ullam et possimus sint fugiat
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 9`] = `""`;

exports[`<FlashMessage /> matches snapshot 10`] = `""`;

exports[`<FlashMessage /> matches snapshot 11`] = `
<styled.div
  className="in eum ut dignissimos aspernatur voluptate voluptatem"
>
  <styled.div>
    <styled.div>
       
      <Link
        href="http://www.Chance.com/"
      >
        Mr. Piper McClure
      </Link>
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 12`] = `
<styled.div>
  <styled.div>
    <styled.div>
      qui repellat assumenda et incidunt aut distinctio
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 13`] = `""`;

exports[`<FlashMessage /> matches snapshot 14`] = `
<styled.div>
  <styled.div>
    <styled.div>
      aut sunt enim rem qui hic dolore
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 15`] = `""`;

exports[`<FlashMessage /> matches snapshot 16`] = `""`;

exports[`<FlashMessage /> matches snapshot 17`] = `""`;

exports[`<FlashMessage /> matches snapshot 18`] = `""`;

exports[`<FlashMessage /> matches snapshot 19`] = `""`;

exports[`<FlashMessage /> matches snapshot 20`] = `""`;

exports[`<FlashMessage /> matches snapshot 21`] = `
<styled.div>
  <styled.div>
    <styled.div>
      non sit a perferendis soluta dignissimos cum
       
      <Link
        href="http://Elyssa.us/"
      >
        Dr. Laurel Balistreri
      </Link>
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 22`] = `
<styled.div
  className="ea sed consequatur ut sapiente molestiae dolores"
>
  <styled.div>
    <styled.div>
      sit optio rerum quasi occaecati modi soluta
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 23`] = `""`;

exports[`<FlashMessage /> matches snapshot 24`] = `
<styled.div
  className="aut non ipsam fuga rem eius est"
>
  <styled.div>
    <styled.div>
      sapiente fuga unde aliquam et debitis dolorem
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 25`] = `
<styled.div>
  <styled.div>
    <styled.div>
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 26`] = `
<styled.div>
  <styled.div>
    <styled.div>
      maiores cumque consequuntur rerum aspernatur nulla aut
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 27`] = `
<styled.div>
  <styled.div>
    <styled.div>
      eum doloribus et perspiciatis sunt eligendi nihil
       
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 28`] = `""`;

exports[`<FlashMessage /> matches snapshot 29`] = `
<styled.div>
  <styled.div>
    <styled.div>
      quisquam voluptatum qui exercitationem eum laudantium autem
       
      <Link
        href="http://Bartell.tv/"
      >
        Mr. Osbaldo Sipes
      </Link>
       
    </styled.div>
    <Styled(Button)
      onClick={[Function]}
    >
      X
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<FlashMessage /> matches snapshot 30`] = `""`;

exports[`<FlashMessage /> matches snapshot 31`] = `""`;

exports[`<FlashMessage /> matches snapshot 32`] = `""`;
